python/src/executor.rs

```diff
@@ -75,6 +75,26 @@ impl BackgroundExecutor {
         rx.recv().unwrap()
     }
 
+    /// Spawn a task in the background
+    pub fn spawn_background<T>(&self, py: Option<Python<'_>>, task: T)
+    where
+        T: Future + Send + 'static,
+        T::Output: Send + 'static,
+    {
+        if let Some(py) = py {
+            py.allow_threads(|| {
+                self.runtime.spawn(task);
+            })
+        } else {
+            // Python::with_gil is a no-op if the GIL is already held by the thread.
+            Python::with_gil(|py| {
+                py.allow_threads(|| {
+                    self.runtime.spawn(task);
+                })
+            })
+        }
+    }
+
```

Comment:
Is this still needed?

Reply:
Yeah that's used in `read_tfrecord`. The pattern I've found for now for exporting streams as recordbatchreaders is to shove the stream on a background task and have it push onto the iterator via a channel. It's a little awkward but seems to work okay.